src/Serilog/Formatting/Display/OutputProperties.cs

```diff
@@ -51,11 +51,17 @@ public static class OutputProperties
         public const string ExceptionPropertyName = "Exception";
 
         /// <summary>
+        /// The properties of the log event.
+        /// </summary>
+        public const string PropertiesPropertyName = "Properties";
+
+        /// <summary>
         /// Create properties from the provided log event.
         /// </summary>
         /// <param name="logEvent">The log event.</param>
+        /// <param name="outputTemplate">The output template.</param>
         /// <returns>A dictionary with properties representing the log event.</returns>
-        public static IReadOnlyDictionary<string, LogEventPropertyValue> GetOutputProperties(LogEvent logEvent)
+        public static IReadOnlyDictionary<string, LogEventPropertyValue> GetOutputProperties(LogEvent logEvent, MessageTemplate outputTemplate)
```

Comment:
Unfortunately, this will break the literate console sink binary:

https://github.com/serilog/serilog-sinks-literate/blob/dev/src/Serilog.Sinks.Literate/Sinks/Literate/LiterateConsoleSink.cs#L88

One obvious option would be to:

 1. Keep this overload
 2. Add back the one without the additional parameter, but make it forward calls to the full version (we could pass `MessageTemplate.Empty` as `outputTemplate`, so that the arguments could still be null-checked)
 3. Mark the old overload `[Obsolete("Pass the full output template using the other overload.")]`

An alternative, which might be worth considering, is to add the new overload as `internal`, mark the old one obsolete, and stop offering a public `GetOutputProperties()` at some time in the future. With the literate console being the only consumer, it may be defensible to simply duplicate the code there.

I like the latter alternative because it'd allow us to re-think how the `MessageTemplateTextFormatter` works in the future - hopefully to improve on the O(N) scans of the message templates - without then needing to break further public API surface.

Any thoughts?